nixos/modules/services/torrent/opentracker.nix

```diff
@@ -3,42 +3,237 @@
 with lib;
 let
   cfg = config.services.opentracker;
+
+  accessListFile = attr:
+    if cfg."${attr}File" != null then
+      cfg."${attr}File"
+    else if cfg.${attr} != null then
+      toString
+        (pkgs.writeText
+          "opentracker-${attr}.txt"
+          (concatMapStrings (hash: "${hash}\n") cfg.${attr}))
+    else
+      null;
+
+  blacklistFile = accessListFile "blacklist";
+  whitelistFile = accessListFile "whitelist";
+
+  overridenPkg = cfg.package.override {
+    blacklist = blacklistFile != null;
+    whitelist = whitelistFile != null;
+    restrictStats = cfg.restrictStats.enable;
+  };
+
+  escapeColon = replaceStrings [ ":" "\\" ] [ "\\:" "\\\\" ];
+
+  mkService = { description, RestrictAddressFamilies, pkg, configFile }: {
+    inherit description;
+    after = [ "network.target" ];
+    requiredBy = [ "opentracker.target" ];
+    partOf = [ "opentracker.target" ];
+    unitConfig.ReloadPropagatedFrom = [ "opentracker.target" ];
+    restartIfChanged = true;
+    serviceConfig = {
+      ExecStart = "${pkg}/bin/opentracker -f ${configFile}";
```

Comment:
cfg.package should be used here

Reply:
No because we must override it to use IPv4 or IPv6.

<https://github.com/NixOS/nixpkgs/blob/6e59ed01b773f54cdad6716b13eed695f9df9465/nixos/modules/services/torrent/opentracker.nix#L21-L25>

<https://github.com/NixOS/nixpkgs/blob/6e59ed01b773f54cdad6716b13eed695f9df9465/nixos/modules/services/torrent/opentracker.nix#L222-L225>